BFWControls/Modules/Transition/View/TranslationAnimationController.swift

```diff
@@ -31,12 +31,13 @@ enum Direction: Int {
 
 }
 
-class TranslationAnimationController: NSObject, UIViewControllerAnimatedTransitioning {
+class TranslationAnimationController: UIPercentDrivenInteractiveTransition, UIViewControllerAnimatedTransitioning, UIViewControllerTransitioningDelegate {
+
 
     // MARK: - Variables
 
     @IBInspectable var isPresenting: Bool = true
-    @IBInspectable var duration: TimeInterval = 0.3
+	@IBInspectable var animateDuration: TimeInterval = 0.3
```

Comment:
As discussed, please rename this `transitionDuration`. I think IBInspectable doesn't like `TimeInterval` (I think it used to, but maybe not since Swift), so please check and if necessary change to `CGFloat`.